dev/mcal/inc/Adc.h

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file Adc.h
+ * @brief 
+ * @version 0.1
+ * @date 2024-07-20
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#ifndef DEV_MCAL_INC_ADC_H_
+#define DEV_MCAL_INC_ADC_H_
+#include "../../mcal/inc/stm32f103xx.h"
+#define ADC_TIMEOUT (400)
+using namespace stm32::registers::adc;
+namespace stm32 {
+namespace dev   {
+namespace mcal  {
+namespace adc   {
+enum  Channel {
```

Comment:
Name it `AdcChannel` to avoid any conflicts with other peripherals that might have the same name.